torch/csrc/jit/passes/tensorexpr_fuser.cpp

```diff
@@ -1182,6 +1194,10 @@ void FuseTensorExprs(
   // Get rid of dead code so that we don't waste effort fusing it.
   EliminateDeadCode(graph);
 
+  if (catOptEnabled()) {
```

Comment:
Is the concat optimization only beneficial when fuser is enabled or does it rely on anything from the fuser? 

If not, can we treat concat optimization as a general graph-level optimization?

Reply:
The second optimization that expands `cat` into copies require shape info. That is the reason I am doing it in the fuser at this point. If we can get shape information by running a shape inference pass separately, then we can move this out of the fuser.

Technically, the `ConcatCse` opt can be done without shape information since that is only looking for common inputs. But that optimization relies on the input lists being materialized before every concat (and that the list operations have been removed which is taken care of by using a traced graph in my perf analysis).

Reply:
OK. Then it makes sense to keep it in the fuser for now.